lib/ansible/module_utils/basic.py

```diff
@@ -167,6 +167,8 @@
 from ansible.module_utils.six.moves import map, reduce
 from ansible.module_utils._text import to_native, to_bytes, to_text
 
+ITERABLETYPE = (map,)
+
```

Comment:
Note: I was thinking more along the lines of:
``` python

# Note: strings are Sequences and all sorts of things (including dictionaries and strings) are iterables.
# So if differentiating between these, it may make a big difference the order that the code checks it in.
# To differentiate those mentioned types, check in this order: [string_types, Sequences, Mappings, Iterables] 
try:
    from collections import iterable, Sequence, Mapping
except ImportError:
    # python2.5
    Sequence = (list, tuple)
    Mapping = (dict,)
    Iterable = (list, tuple, map)
``` 

Thinking about this even more, I think you can probably get rid of SEQUENCETYPE altogether as long as the places where it should accept a KeyView should be taking Iterable instead (KeyView is actually an iterable but not a Sequence.  I think the current code is abusing that.)